### DISCUSSION TITLE
using drivebase as a sensor?

### DISCUSSION THREAD

**Discussion by jediforri** [pybricks/support#1682] [General]
Is there a way to use drivebase and measure when the wheels are having trouble (e.g. they have hit a wall)?
As drivebase_instance.straight() is synchronous, I am not sure how to measure the wheel motors when in this call.
Thank you!

**Comment by laurensvalk** [comment]
Are you using Python coding? You can set the `wait` parameter in `straight` to `False`, so that your code continues without waiting for the movement to finish. Here is an [example for single motors](https://docs.pybricks.com/en/latest/pupdevices/motor.html#waiting-for-two-parallel-actions-to-complete) - it is similar for drive bases.

Instead of wait for two motors to complete as in that example, you could wait until `drivebase_instance.done()` as well as something like the time being done, or perhaps the wheel speed being slowed down. And then decide to stop.
